lib/utils.js

```diff
@@ -56,3 +56,13 @@ exports.deepFilterByKey = function (obj, cb) {
     return clean
   }, {})
 }
+
+/**
+ * Given a triple returns a collection of all subjects that occur as initial ' -- '-delimited substrings of the subject
```

Comment:
I am assuming some have only two portions divided by ` -- `? Also in the future, we could have more than three?

Reply:
Definitely some have only two portions. As far as I know it could be more than three